### PR TITLE
Add GitHub Actions checks and tests

### PR DESCRIPTION
**Pull request by InnovativeInventor**
There are some errors rn, so this is a draft PR.

### CODE REVIEW

**Comment by InnovativeInventor** [comment]
NB: The format tests are supposed to fail. Right before we merge, I'll pull in the latest commit and run the formatter so we don't get any nasty merge conflicts.

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+on: [push, pull_request]
+
+jobs:
+  test: # from the docs
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        julia-version: ['1.5.3']
+        os: [ubuntu-latest]
```
Comment (bsuwal) [inline .github/workflows/test.yml:11]: Could you also add `macos-latest` and `windows-latest`?

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+      name: "Build pkg"
+    - uses: julia-actions/julia-runtest@v1
+      name: "Run tests"
+
+  format:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        julia-version: ['1.5.3']
+        os: [ubuntu-latest]
```
Comment (bsuwal) [inline .github/workflows/test.yml:31]: I'm fine with the docs build checking on only `ubuntu`

File: src/scores.jl [hunk 3]
```diff
@@ -746,7 +746,7 @@ function save_scores_to_hdf5(filename::String,
                 end
                 # create the initial datasets
                 if idx == 1
-                    scores_datasets[key] = d_create(
+                    scores_datasets[key] = create_dataset(
```
Comment (bsuwal) [inline src/scores.jl:749]: thanks!

**Review by bsuwal — APPROVED** [review]
Looks good to me! Could you also open an issue about removing us from Travis? Also I suggested that we also run builds on `windows` and `macos. Excited for your first merge!

**Comment by bsuwal** [comment]
@InnovativeInventor another thing -- I just noticed that you included tests against the `nightly` Julia build -- I would recommend removing that. Nightly builds are unstable and often times the tests fail for them - at the very least you should allow them to fail but at  that point I find it cleaner to drop the `nightly` tests completely.

**Pull request merged**